lib/folio_sync/archives_space_to_folio/job_result_processor.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module FolioSync
+  module ArchivesSpaceToFolio
+    class JobResultProcessor
+      attr_reader :processing_errors
+
+      def initialize(folio_reader, folio_writer, instance_key)
+        @folio_reader = folio_reader
+        @folio_writer = folio_writer
+        @instance_key = instance_key
+        @processing_errors = []
+      end
+
+      # Processes job execution results, handling both suppression updates and database record updates
+      # @param job_execution_summary [Folio::Client::JobExecutionSummary] The completed job summary
+      # @return [Array<FolioSync::Errors::SyncingError>] Any processing errors that occurred
+      def process_results(job_execution_summary)
+        job_execution_summary.each_result do |_raw_result, custom_metadata, instance_action_status, hrid_list, id_list|
+          # Update suppression status for successful records
+          update_suppression_status(custom_metadata, id_list) if ['CREATED', 'UPDATED'].include?(instance_action_status)
+
+          # Update database record status
+          update_database_record(custom_metadata, instance_action_status, hrid_list)
+        end
+      end
+
+      private
+
+      def update_suppression_status(custom_metadata, id_list)
+        return if id_list.blank?
+
+        instance_record_id = id_list.first
+        incoming_suppress = custom_metadata[:suppress_discovery]
+
+        begin
+          folio_record = @folio_reader.get_instance_by_id(instance_record_id)
+          current_folio_suppress = folio_record['discoverySuppress']
+
+          unless current_folio_suppress != incoming_suppress
```

Comment:
This is just personal preference, but this might be slightly easier to read:
```suggestion
          if current_folio_suppress == incoming_suppress
```